remotes/docker/fetcher.go

```diff
@@ -152,6 +154,7 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 
 func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
 	req.header.Set("Accept", strings.Join([]string{mediatype, `*/*`}, ", "))
+	req.header.Set("Accept-Encoding", "zstd, gzip")
```

Comment:
> that effectively was the "previous"
nope, net/http automatically set `Accept-Encoding: gzip` if you don't set this header on your own

Abut weight, I don't think fetcher has any preference? Registry probably has, because it might already host/cache pre-compressed entities

Reply:
> nope, net/http automatically set Accept-Encoding: gzip if you don't set this header on your own

Ah, thanks!

> About weight, I don't think fetcher has any preference? Registry probably has, because it might already host/cache pre-compressed entities

Yes, so for these

- the Client sends a list of what it accepts, and a preferred order (so in this case, may be `zstd` )
- if the _server_ supports multiple of them, it will then pick the one that the client indicated to be the preferred one.

I _think_ if no weights are set, that it's mostly "undefined" (so we _may_ get `gzip` or we _may_ get `zstd`, even if the registry would support both); https://developer.mozilla.org/en-US/docs/Glossary/Quality_values

> Quality values, or q-values and q-factors, are used to describe the order of priority of values in a comma-separated list
...
> If there is no priority defined for the first two values, the order in the list is irrelevant.